pkg/controller/pxc/controller.go

```diff
@@ -1109,3 +1099,40 @@ func createOrUpdateConfigmap(cl client.Client, configMap *corev1.ConfigMap) erro
 
 	return nil
 }
+
+func updateServiceIfNeeded(cl client.Client, old, new *corev1.Service) error {
+	if !isServicesSpecEqual(&old.Spec, &new.Spec) || !reflect.DeepEqual(old.Annotations, new.Annotations) ||
+		!reflect.DeepEqual(old.Labels, new.Labels) || !reflect.DeepEqual(new.Spec.Selector, old.Spec.Selector) {
+		log.Info("UPDATING HAPROXY", "name", new.Name)
+		return cl.Update(context.TODO(), new)
+	}
+	return nil
+}
+
+func isServicesSpecEqual(old, new *corev1.ServiceSpec) bool {
+	if old.Type != new.Type || !isServicesPortsEqual(old.Ports, new.Ports) ||
+		old.ExternalTrafficPolicy != new.ExternalTrafficPolicy ||
+		!reflect.DeepEqual(old.LoadBalancerSourceRanges, new.LoadBalancerSourceRanges) {
```

Comment:
It's easy co compare 2 slices without using reflection

Reply:
unfortunately, there are other fields set by k8s which we need to ignore)
That's why i have to do it manually

Reply:
I ment we can sort them and compare index by index